src/components/WeatherIcon.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { View, Image, StyleSheet } from 'react-native';
+
+export enum WeatherType {
+    Sunny,
+    Thunderstorm,
+}
+
+interface IImageData {
+    src: number;
+    w: number;
+    h: number;
+}
+
+type WeatherToImage = {
+    [key in WeatherType]: IImageData;
+};
+
+const Images: WeatherToImage = {
+    [WeatherType.Sunny]: {
+        src: require('@assets/icons/sun.png'),
+        w: 35,
+        h: 35,
+    },
+    [WeatherType.Thunderstorm]: {
+        src: require('@assets/icons/thunderstorm.png'),
+        w: 42,
+        h: 33,
+    },
+};
+
+export const WeatherIcon: React.FC<{ weatherType: WeatherType }> = props => {
+    const image = Images[props.weatherType];
+    return (
+        <View style={styles.wrapper}>
+            <Image source={image.src} style={{ width: image.w, height: image.h }} />
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
+    wrapper: {
+        weight: '20%',
+        height: '20%',
+    },
```

Comment:
For images width and heigh it is better to use absolute values, or values based on device screen width, when you use % it will base on parent width, which might be different
![photo_2023-03-10_14-48-29](https://user-images.githubusercontent.com/52620425/224308608-bb034d7a-5ab4-44ab-b5a9-34d8d7531ad9.jpg)